src/aviary/dataset_server.py

```diff
@@ -0,0 +1,210 @@
+import asyncio
+import logging
+import time
+import traceback
+import uuid
+from contextlib import contextmanager
+from itertools import starmap
+from typing import Generic, TypeVar
+
+import uvicorn
+from pydantic import BaseModel, Field
+
+from aviary.env import Environment, TaskDataset
+from aviary.tools import MessagesAdapter, ToolRequestMessage, ToolsAdapter
+
+try:
+    from fastapi import FastAPI, HTTPException
+except ImportError:
+    # We will raise if a TaskDatasetServer is instantiated but fastapi is not available
+    FastAPI = HTTPException = None
+
+logger = logging.getLogger(__name__)
+
+
+class StartRequest(BaseModel):
+    task: int | None = None
```

Comment:
Is this a task ID? if yes, can we have it named as such? Or can you add a description saying what the `task` means

Reply:
It's a task index. Will rename